app/controllers/employments/EndEmploymentController.scala

```diff
@@ -99,15 +99,10 @@ class EndEmploymentController @Inject()(
 
   private def redirectToWarningOrDecisionPage(
     journeyCacheFuture: Future[Map[String, String]],
-    successfullJourneyCacheFuture: Future[Option[String]])(implicit hc: HeaderCarrier): Future[Result] =
-    for {
-      _                      <- journeyCacheFuture
-      successfulJourneyCache <- successfullJourneyCacheFuture
-    } yield {
-      successfulJourneyCache match {
-        case Some(_) => Redirect(routes.EndEmploymentController.duplicateSubmissionWarning())
-        case _       => Redirect(routes.EndEmploymentController.employmentUpdateRemoveDecision())
-      }
+    successfulJourneyCacheFuture: Future[Option[String]])(implicit hc: HeaderCarrier): Future[Result] =
```

Comment:
because of the type signature the futures u r passing in are already concurrent as the types are not deferred (`: =>`)